tfs/src/main/java/hudson/plugins/tfs/TeamPushTrigger.java

```diff
@@ -127,36 +131,54 @@ public void run() {
                     changesDetected = "SCM changes detected in " + job.getFullDisplayName() + ". ";
                     shouldSchedule = true;
                 }
-            }
-            else {
+            } else {
                 changesDetected = "Polling bypassed for " + job.getFullDisplayName() + ". ";
             }
             if (shouldSchedule) {
                 final SCMTriggerItem p = job();
                 final String name = "#" + p.getNextBuildNumber();
                 final String pushedBy = gitCodePushedEventArgs.pushedBy;
+
+                String archGroup = "";
+                String configGroup = "";
+                if (p instanceof WorkflowJob) {
+                    final List<JobProperty<? super WorkflowJob>> jobProperties = ((WorkflowJob) p).getAllProperties();
+                    for (final JobProperty<? super WorkflowJob> jobProperty : ((WorkflowJob) p).getAllProperties()) {
+                        if (jobProperty instanceof ParametersDefinitionProperty) {
+                            ParametersDefinitionProperty pDefProp = (ParametersDefinitionProperty) jobProperty;
+                            for (final ParameterDefinition currParam : pDefProp.getParameterDefinitions()) {
+                                final String paramName = currParam.getName();
+                                if (paramName.equalsIgnoreCase("CGroup") || paramName.equalsIgnoreCase("Configuration")) {

```

Comment:
CGroup and AGroup are just names that someone invented on our side, why are we using these here?

Reply:
Had a offline discussion with Matt and working on improving the UI as he suggested.